src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,115 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import { useEffect } from 'react';
+
+import { getNumbers, getList } from '../../utils';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+  onItemChange: (el1: number[]) => void
+}
+
+export const Pagination = (
+  {
+    total,
+    perPage,
+    currentPage,
+    onPageChange,
+    onItemChange,
+  }: Props,
+) => {
+  const amountPages = total % perPage === 0
+    ? Math.floor(total / perPage)
+    : Math.floor(total / perPage) + 1;
+
+  function pageHendler(event: React.MouseEvent<HTMLAnchorElement>) {
+    const selectPage = +event.currentTarget.innerText;
+
+    onPageChange(selectPage);
+  }
+
+  const pages = getNumbers(1, amountPages);
+
+  const amountOfli = getList(currentPage, total, perPage);
+
+  const [start, end] = amountOfli;
+
+  const items = getNumbers(start, end).map(n => `Item ${n}`);
+
+  function arrowHendlerLeft(event: React.MouseEvent<HTMLElement>) {
+    if (event && currentPage > pages[0]) {
+      onPageChange(currentPage - 1);
+    }
+  }
+
+  function arrowHendlerRight(event: React.MouseEvent<HTMLElement>) {
+    if (event && currentPage < amountPages) {
+      onPageChange(currentPage + 1);
+    }
+  }
+
+  useEffect(() => {
+    onItemChange(amountOfli);
+  }, [currentPage, perPage]);
+
+  return (
+    <>
```

Comment:
this fragment wrapper is redundant
```suggestion
```

Reply:
https://github.com/mate-academy/react_pagination/pull/654/files#r1346282934
need to fix it

Reply:
There are two tags of <ul><ul/>, that is why I use i<></>.